src/lib/cinchy.service.ts

```diff
@@ -35,10 +56,36 @@ export class CinchyService {
                 this.refreshTokenOnLoadIfNeeded();
             }
         }
+
+        // After the router resolves, it's probable that the authentication workflow has removed the querystring,
+        // so if we have one stored, we need to add it back in
+        this._router.events.subscribe({
```

Comment:
Shouldn't this be subscribed and triggered once right after successful login redirect from Cinchy? 

In the scenario where the user is already authenticated, I think it may update the query params upon the first routing and cause unexpected behaviour. Example:

(assume authenticated)
1. User enters app with query parameters
2. User uses CinchyService, peforms whatever query
3. User presses a button that routes them to another path within the app

In this scenario, would the query parameters persist to the next route after step 3?